processors/ratelimiter/ratelimiterprocessor.go

```diff
@@ -0,0 +1,125 @@
+package ratelimiter
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	pb_struct "github.com/envoyproxy/go-control-plane/envoy/extensions/common/ratelimit/v3"
+	pb "github.com/envoyproxy/go-control-plane/envoy/service/ratelimit/v3"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/tag"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+var _ component.TracesProcessor = (*processor)(nil)
+
+func (p *processor) Start(_ context.Context, _ component.Host) error {
+	return nil
+}
+
+func (p *processor) Shutdown(_ context.Context) error {
+	p.rateLimitServiceClientConn.Close()
```

Comment:
These close and cancel functions do not return errors? If they do we should not ignore them. We should log them.

Reply:
Fixed it.